### PR TITLE
Update pipeline to 1ES compliant template

### PR DESCRIPTION
**Pull request by noahdarveau-MSFT**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

> Summarize the changes, including the goals and reasons for this change. Be sure to call out any technical or behavior changes that reviewers should be aware of.

> If this Pull Request should close/resolve any issues when merged, use [the special syntax for that](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) here.

### Main changes in the PR:

1. Migrating to the 1ES compliant pipeline template.

### Validation performed:

1. <Step 1>
2. <Step 2>

### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

<Yes/No>

### End-to-end tests added:

<Yes/No>

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

<Yes/No>

### Related PRs:

> Remove this section if n/a

### Next/remaining steps:

> List the next or remaining steps in implementing the overall feature in subsequent PRs (or is the feature 100% complete after this?).

> Remove this section if n/a

- [ ] Item 1
- [ ] Item 2

### Screenshots:

> Remove this section if n/a

| Before     | After      |
| ---------- | ---------- |
| < image1 > | < image2 > |

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,162 +1,551 @@
...
+        customEnvironments: true
+        environmentsBrowser: true
+        environmentsNode: true
+        environmentsCommonJs: true
+        environmentsSharedNodeBrowser: false
+        environmentsEs6: false
+        environmentsEs2017: false
+        environmentsEs2020: true
+        environmentsJest: false
+      # sourceAnalysisPool:
```
Comment (noahdarveau-MSFT) [inline azure-pipelines.yml:48]: This seems unnecessary. Going to remove it for now

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,162 +1,551 @@
...
+                  artifactName: 'CDNFeed'
+                - output: pipelineArtifact
+                  displayName: 'Publish NPM feed to Build Artifacts'
+                  targetPath: '$(Build.ArtifactStagingDirectory)\NPMFeed'
+                  artifactName: 'NPMFeed'
+                - output: pipelineArtifact
+                  displayName: 'Publish Powershell Scripts to Build Artifacts'
+                  targetPath: '$(Build.ArtifactStagingDirectory)\scripts'
+                  artifactName: 'scripts'
+                # - output: pipelineArtifact
```
Comment (noahdarveau-MSFT) [inline azure-pipelines.yml:258]: Not sure why we this is here. I don't know if I fully understand SBOM, but it seems it's essentially trying to make a manifest out of the manifest and publish it? It also doesn't not work, so commenting it out for now.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -1,162 +1,580 @@
...
+                  testResultsFormat: 'JUnit'
+                  testResultsFiles: '**/e2e-tests-report*.xml'
+                  testRunTitle: 'E2E Tests - Perf'
+                  mergeTestResults: true
+                condition: succeededOrFailed()
+          - job: E2ETestAndroid
+            displayName: 'E2E Test - Android'
+            pool:
+              name: Azure Pipelines
+              image: internal-macos-11
```
Comment (noahdarveau-MSFT) [inline azure-pipelines.yml:404]: Can this be updated to macos-12?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -1,162 +1,583 @@
...
+    - repository: AppHostingSdk
+      type: git
+      name: ISS/metaos-hub-sdk
+      ref: refs/heads/main
+    #- repository: AndroidAppHostingSdk
+    #  type: git
+    #  name: ISS/metaos-hub-sdk-android
+    - repository: IOSAppHostingSdk
+      type: git
+      name: ISS/metaos-hub-sdk-ios
```
Comment (KangxuanYe) [inline azure-pipelines.yml:47]: Leaking sensitive info. Please remove or use variable instead.

Comment (KangxuanYe) [inline azure-pipelines.yml:47]: Same to these places.

File: azure-pipelines.yml (hunk 3 above)
Comment (KangxuanYe) [reply azure-pipelines.yml:404]: I think we can. What do you think? @jekloudaMSFT

**Review by KangxuanYe — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml [hunk 5]
```diff
@@ -1,162 +1,593 @@
...
+                inputs:
+                  script: |
+                    pnpm build-force-blazor
+                  workingDirectory: '$(ClientSdkProjectDirectory)'
+              - task: CmdLine@2
+                displayName: 'Configure host machine'
+                inputs:
+                  script: "sudo chmod -R 755 ./ \npnpm run setup\n"
+                  workingDirectory: '$(AppHostingSdkProjectDirectory)'
+              - task: CmdLine@2
```
Comment (noahdarveau-MSFT) [inline azure-pipelines.yml:391]: Was using this for debugging. Needs to be removed before check-in

**Review by jadahiya-MSFT — APPROVED** [review]

File: azure-pipelines.yml [hunk 6]
```diff
@@ -4,159 +4,551 @@ variables:
...
+                displayName: 'Disable Slow Animation'
+                inputs:
+                  targetType: inline
+                  script: cd ~/Library/Preferences defaults write com.apple.iphonesimulator SlowMotionAnimation -bool NO
+                  workingDirectory: '$(System.DefaultWorkingDirectory)'
+              - task: Bash@3
+                displayName: 'iOS UI/E2E Tests'
+                inputs:
+                  targetType: inline
+                  script: "/usr/bin/xcodebuild -configuration Release -workspace /Users/runner/work/1/iOSHost/$(IOSSdkWorkspace).xcworkspace -scheme $(IOSSdkScheme) -sdk iphonesimulator -parallel-testing-enabled YES -parallel-testing-worker-count 2 -destination 'platform=iOS Simulator,name=iPhone 14 Pro,OS=16.2' -screenshot-enabled=\"YES\" -test-iterations 3 -retry-tests-on-failure -quiet -resultBundlePath TestResults test > report.out 2>&1\nif [[ $? == 0 ]]; then echo \"E2E Test passes successfully\"; exit 0; else echo \"E2E Test failed\"; exit 1; fi;"
```
Comment (KangxuanYe) [inline azure-pipelines.yml:536]: could you please double check on test-iteration and retry-tests-on-failure parameters? I believe I have removed them. And also I am pretty sure there is no report.out file.

Please check here: https://github.com/OfficeDev/microsoft-teams-library-js/pull/1995/files

**Review by KangxuanYe — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml [hunk 7]
```diff
@@ -4,159 +4,551 @@ variables:
...
+                  workingDirectory: '$(Agent.BuildDirectory)/iOSHost'
+              - task: Bash@3
+                displayName: 'Generate E2E test report'
+                condition: succeededOrFailed()
+                inputs:
+                  targetType: inline
+                  script: |
+                    xchtmlreport -r TestResults -j > /dev/null; if [[ $? == 0 ]]; then echo "Test report has been generated successfully."; exit 0; else echo "Test report generating process failed for some reasons."; exit 1; fi;
```
Comment (KangxuanYe) [inline azure-pipelines.yml:544]: Same here.

**Review by KangxuanYe — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml [hunk 8]
```diff
@@ -4,159 +4,551 @@ variables:
...
+                    pnpm build-force-blazor
+                  workingDirectory: '$(ClientSdkProjectDirectory)'
+              - task: Bash@3
+                displayName: 'Run sample test app in background'
+                inputs:
+                  targetType: inline
+                  script: 'nohup pnpm start-test-app &'
+                  workingDirectory: '$(ClientSdkProjectDirectory)'
+              - task: Bash@3
+                displayName: 'Install Emulator'
```
Comment (jekloudaMSFT) [inline azure-pipelines.yml:424]: Can you copy over the 10 minute timeout on this task? If it goes longer than that, it almost always times out after the whole 60 minutes.

File: azure-pipelines.yml [hunk 9]
```diff
@@ -4,159 +4,551 @@ variables:
...
+                displayName: 'Install client sdk dependencies'
+                inputs:
+                  script: |
+                    pnpm install
+                  workingDirectory: '$(ClientSdkProjectDirectory)'
+              - task: CmdLine@2
+                displayName: 'Build client sdk'
+                inputs:
+                  script: |
+                    pnpm build-force-blazor
```
Comment (jekloudaMSFT) [inline azure-pipelines.yml:415]: Is this task building the blazor app?

File: azure-pipelines.yml [hunk 10]
```diff
@@ -4,159 +4,551 @@ variables:
...
+                  workingDirectory: '$(ClientSdkProjectDirectory)'
+              - task: Bash@3
+                displayName: 'Install Emulator'
+                inputs:
+                  targetType: inline
+                  script: 'chmod u+x install_emulator.sh && ./install_emulator.sh'
+                  workingDirectory: '$(Agent.BuildDirectory)/androidHost/devtools/ci'
+              - task: JavaToolInstaller@0
+                inputs:
+                  versionSpec: '11'
```
Comment (jekloudaMSFT) [inline azure-pipelines.yml:431]: Can you bump this to version 17?

File: azure-pipelines.yml (hunk 9 above)
Comment (noahdarveau-MSFT) [reply azure-pipelines.yml:415]: It's building everything, blazor app included

File: azure-pipelines.yml (hunk 6 above)
Comment (noahdarveau-MSFT) [reply azure-pipelines.yml]: Fixed

File: azure-pipelines.yml (hunk 7 above)
Comment (noahdarveau-MSFT) [reply azure-pipelines.yml]: Fixed

File: azure-pipelines.yml (hunk 10 above)
Comment (noahdarveau-MSFT) [reply azure-pipelines.yml]: Done

**Review by KangxuanYe — APPROVED** [review]
LGTM. Approved for iOS.

**Review by jekloudaMSFT — APPROVED** [review]

**Pull request merged**